docs/spec/lightclient/detection/detection.md

```diff
@@ -363,8 +560,92 @@ The detector shall maintain the following invariants:
 and the following transition invariant
    - *FullNodes' \union Secondaries' \union FaultyNodes' = FullNodes \union Secondaries \union FaultyNodes*
 
+
+> The following invariant is very useful for reasoning, and underlies
+> many intuition when we 
+
+#### **[LCD-INV-TRUSTED-AGREED.1]:**
+
+It is always the case the light client has downloaded a lightblock for height
+*lightStore.LatestTrusted().Height* 
+from each of the current primary and the secondary, that all reported
+the identical lightblock for that height.
+
+> In the above, I guess "the identical" might be replaced with "a
+> matching" to cover commits that might be different.
+
+> The above requires us that before we pick a new secondary, we have to
+> query the secondary for the header of height 
+> *lightStore.LatestTrusted().Height*.
+
+
 ## Solution
 
+### Data Structures
+
+Lightblocks and LightStores are
+defined at [LCV-DATA-LIGHTBLOCK.1] and [LCV-DATA-LIGHTSTORE.1]. See the [verification specification][verification] for details.
+
+> The following data structure [LCV-DATA-POF.1]
+> defines a **proof of fork**. Following
+> [TMBC-SIGN-FORK.1], we require two blocks *b* and *c* for the same
+> height that can both be verified from a common root block *a* (using
+> the skipping or the sequential method).
+
+> [LCV-DATA-POF.1] mirrors the definition [TMBC-SIGN-FORK.1]:
+> *TrustedBlock* corresponds to *a*, and *PrimaryTrace* and *SecondaryTrace*
+> are traces to two blocks *b* and *c*. The traces establish that both
+> *skip-root(a,b,t)* and *skip-root(a,c,t)* are satisfied.
```

Comment:
Do we need to enforce that the heights used in verification via skipping must be the same for both `skip-root(a,b,t)` and `skip-root(a,c,t)`

Reply:
No. Just the blocks *b* and *c* need to be the same height. The other blocks on the traces need not be at the same height. Due to the dynamic changes in the validator set, in general skipping via the same heights might not work on different branches. 

Actually, if there would be earlier two conflicting blocks with the same height on two different traces, we could use those earlier blocks as proof of fork. Therefore, attackers will most likely try to avoid earlier conflicting block.